sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java

```diff
@@ -86,9 +86,20 @@
       BigQueryOptions bqOptions = options.as(BigQueryOptions.class);
       TableReference tableToExtract = getTableToExtract(bqOptions);
       JobService jobService = bqServices.getJobService(bqOptions);
+
+      final String extractDestinationDir;
+      String tempLocation = bqOptions.getTempLocation();
+      try {
```

Comment:
This repeats 5 times throughout the PR, it's way too much. Extract into a utility? (e.g. into BigQueryHelpers)